unicodetools/src/main/java/org/unicode/text/UCD/VersionedSymbolTable.java

```diff
@@ -0,0 +1,426 @@
+                            UcdPropertyValues.General_Category_Values.Symbol,
+                            new String[] {"Sc", "Sk", "Sm", "So"},
+                            UcdPropertyValues.General_Category_Values.Separator,
+                            new String[] {"Zl", "Zp", "Zs"});
+
+    /**
+     * Similar to iup.getProperty(UcdProperty.General_Category).getSet(propertyValue), but takes the
+     * groupings into account. Implements both unary-query-expression for a General_Category alias and
+     * binary-query-expression with a property-value where the queried property is General_Category.
+     */
+    private UnicodeSet getGeneralCategorySet(IndexUnicodeProperties iup, String propertyValue) {
+        var gc = iup.getProperty(UcdProperty.General_Category);
+        for (var entry : COARSE_GENERAL_CATEGORIES.entrySet()) {
+            final var aliases = entry.getKey().getNames().getAllNames();
+            if (aliases.stream().anyMatch(a -> UnicodeProperty.equalNames(propertyValue, a))) {
+                UnicodeSet result = new UnicodeSet();
+                for (var value : entry.getValue()) {
+                    gc.getSet(value, result);
+                }
+                return result;
+            }
+        }
+        return gc.getSet(propertyValue);
+    }
+
+    @Override
+    public boolean applyPropertyAlias(String beforeEquals, String afterEquals, UnicodeSet result) {
+        result.clear();
+        String leftHandSide = beforeEquals;
+        String propertyPredicate = afterEquals;
+        boolean interiorlyNegated = false;
+        int posNotEqual = beforeEquals.indexOf('≠');
+        // TODO(egg): We cannot distinguish \p{X=} from \p{X} in this API, both give us an empty
+        // string as afterEquals.  This is an @internal API, so we could change it to pass null in
+        // the unary case.
+        if (posNotEqual >= 0) {
+            propertyPredicate =
+                    afterEquals.length() == 0
+                            ? beforeEquals.substring(posNotEqual + 1)
+                            : beforeEquals.substring(posNotEqual + 1) + "=" + afterEquals;
+            leftHandSide = beforeEquals.substring(0, posNotEqual);
+            interiorlyNegated = true;
+        }
+        if (interiorlyNegated) {
+            final var complement = getNonNegatedPropertyQuerySet(leftHandSide, propertyPredicate);
+            result.addAll(complement.complement().removeAllStrings());
+        } else {
+            result.addAll(getNonNegatedPropertyQuerySet(leftHandSide, propertyPredicate));
+        }
+        return true;
+    }
+
+    private UnicodeSet getNonNegatedPropertyQuerySet(
+            String leftHandSide, String propertyPredicate) {
+        final var unqualifiedLeftHandSide = new StringBuilder(leftHandSide);
+        final var queriedVersion = parseVersionQualifier(unqualifiedLeftHandSide);
+        final var deducedQueriedVersion = queriedVersion == null ? implicitVersion : queriedVersion;
```

Comment:
totally optional: some of the variable names seem longer than necessary. for example, this one i would just call "version". for "queriedProperties", maybe just "properties". etc.

shorter names make for easier reading, shorter lines, and less wrapping. just not so short that they get confusing :-)

Reply:
I get your point, and I think I would normally have called this `version` rather than `queriedVersion`, but since this is written against the UnicodeSet spec, I am trying to match the terminology there. And since there I need to talk about two versions, they both have qualified names.